langchain4j-azure-ai-search/src/main/java/dev/langchain4j/rag/content/retriever/azure/search/AzureAiSearchContentRetriever.java

```diff
@@ -182,7 +196,8 @@ private List<Content> findRelevantWithFullText(String content, int maxResults, d
         SearchPagedIterable searchResults =
                 searchClient.search(content,
                         new SearchOptions()
-                                .setTop(maxResults),
+                                .setTop(maxResults)
+                                .setFilter(filterMapper.map(filter)),
```

Comment:
minor: `Filter` can be mapped in the ctor (once), so there would be no need to map it on every search